extensions/vscode/src/extension/parseUsesSlug.ts

```diff
@@ -0,0 +1,53 @@
+const USES_PATTERN_STRING = "\\buses:\\s*(.+)$";
+
+export function parseUsesSlug(
+  lineText: string,
+  position?: number,
+): string | undefined {
+  const [slug, index] = findSlug(lineText, position);
+  if (slug) {
+    return slug;
+  }
+  return undefined;
+}
+
+export function findSlug(
+  lineText: string,
+  position?: number,
+): [string | undefined, number | undefined] {
+  const pattern = new RegExp(USES_PATTERN_STRING, "dg");
+  let match;
+  while ((match = pattern.exec(lineText)) !== null) {
+    let slug = match[1].trim();
+    let quoteOffset = 0;
+
+    // Check for surrounding quotes
+    const quoteMatch = slug.match(/^(['"])(.*)\1/);
```

Comment:
Minor point: this should be `.*?\1` so that `"$granite-code/autocomplete" # Use " just because` doesn't match all the way up to the second "

Reply:
nice catch! TBH, I suck at regexps ;-)